clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -615,6 +629,7 @@ public ObjectLocation pathToObjectLocation(Path path) {
     }
 
     class ListingIterator implements RemoteIterator<LakeFSLocatedFileStatus> {
+        private final boolean recursive;
```

Comment:
Please rename this parameter for what it does on the iterator.  LFSFS _uses_ it to implement "recursive" listings. But AFAICT what it actually _does_ in the iterator -- which is created after the LakeFS API call has already been made with the "recursive" flag -- is remove "directories" from the list.  Maybe it should be called "removeDirectories"?